### ISSUE TITLE
Integrate EIA-860 Multifuel table

### ISSUE THREAD

**Issue by grgmiller** [catalyst-cooperative/pudl#3438]
### Overview

3_5_Multifuel_Y#### is a table in EIA-860 that includes detailed information about multifuel generators. 

Why do you want it in PUDL?

Some of the columns in the multifuel table are in pudl from other EIA-860 tables, but there are columns in the multifuel table that are unique to that table (for example, the time to switch from oil to gas), and the data in this table seems more complete than the duplicate columns in other tables (for example, this table indicates that plant 10 can switch fuels while operating (`operating_switch = "Y"`, but the data for this plant currently integrated into pudl has this as a missing value. 

### Logistics

This is public and published alongside all of the other EIA-860 tables. Integrating may require reconciling overlapping columns from other EIA-860 tables if they disagree with each other. 

### What do you know about it so far?

It would be great to have this to ensure that `multiple_fuels`, 'cofire_fuels`, `switch_oil_gas`, and `operating_switch` are accurate and complete. 

It would also be helpful to have some new data such as the specific breakout of different cofire energy sources (as distinct from `energy_source_code` and `startup_source_code`); the time to switch columns, and the factors that limit switching columns.

**Comment by zaneselvans** [comment]
I think we have concrete plans to get all of the other tables from schedule 3, but hadn't planned for this one yet. Would you be up for making a PR that compiles the necessary file/sheet/column mappings? And then maybe we could work on the transform alongside the other new EIA-860 tables.

**Comment by zaneselvans** [comment]
@grgmiller I'm going to flesh out this issue a bit in the hopes that we can make it accessible to an open source contributor.

**Comment by grgmiller** [comment]
I don't think this is a complete list, but here's some of the columns mappings that we've been parsing for our uses:
```
column_names = {
        "Plant Code": "plant_id_eia",
        "Generator ID": "generator_id",
        "Energy Source 1": "energy_source_code_1",
        "Energy Source 2": "energy_source_code_2",
        "Multiple Fuels?": "multiple_fuels",
        "Cofire Fuels?": "cofire_fuels",
        "Cofire Energy Source 1": "cofire_source_code_1",
        "Cofire Energy Source 2": "cofire_source_code_2",
        "Cofire Energy Source 3": "cofire_source_code_3",
        "Cofire Energy Source 4": "cofire_source_code_4",
        "Cofire Energy Source 5": "cofire_source_code_5",
        "Cofire Energy Source 6": "cofire_source_code_6",
        "Switch Between Oil and Natural Gas?": "switch_oil_gas",
        "Switch When Operating?": "operating_switch",
        "Time to Switch From Gas to Oil": "switch_gas_oil_time",
        "Time to Switch From Oil to Gas": "switch_oil_gas_time",
        "Factors that Limit Switching?": "switch_limits",
        "Air Permit Limits?": "air_permit_limit",
    }
```

**Comment by jmelot** [comment]
Hi, just came across this project. I have some free time next week and would be interested in helping out with this issue, if it's still needed. Would looking through https://github.com/catalyst-cooperative/pudl/blob/main/src/pudl/transform/eia860.py be a good place to start?

**Comment by zaneselvans** [comment]
Hey @jmelot I think the first thing that needs to happen is the spreadsheet tab needs to be mapped for extraction.  You can see a bunch of other examples of how we do that under `src/pudl/package_data/eia860/`.  You'll need to set up [the development environment](https://catalystcoop-pudl.readthedocs.io/en/nightly/dev/dev_setup.html) and pull down the EIA-860 input data archives with `pudl_datastore --datasets eia860` so you can inspect all the historical spreadsheets to see what columns are called, how many rows to skip for headers, etc.

Then the tab can get added to the raw extraction step in `src/pudl/extract/eia860.py` and you can try running the extraction in Dagster and inspect the resulting raw dataframe. Then you could move on to the transform step.

**Comment by jmelot** [comment]
Makes sense, thanks for the help getting started!

**Comment by grgmiller** [comment]
Thanks, appreciate that you're interested in tackling this table! Feel free to reach out if you have any questions about the table since I've spent a bit of time looking at it and using data from this table.

**Comment by jmelot** [comment]
It looks like the mappings under `src/pudl/package_data/eia860/` are already present, except that there are only column maps for the `existing` and `retired` tabs. Should I add a map for the `proposed` tab as well?

Thanks for all the clear instructions, it was pretty smooth getting things set up! I'm waiting on the `ferc_to_sqlite` job to finish and then I'll try writing a transform.

**Comment by zaneselvans** [comment]
Oh great! I didn't realize we'd gotten it to the point of extraction already. If there's data in the proposed tab that would be good to get mapped too -- the other generators also come in those three flavors and it would be nice to have the same coverage for all of them.

**Comment by aesharpe** [comment]
Resolved in #3988

**Comment by aesharpe** [comment]
Resolved in #3988
